File: PW/ConfigEditor/src/Structures/Type.java
package Structures;

import Core.Main;

public class Type
{
    public String _name,_type;
    public int _size;

    public Type(String name, String type)
    {
        _name = name;
        _type = type;
        if(!_type.contains(":")) {
            Main.All_Types.add(_type);
            switch (_type) {
                case "int32":
                    _size = 4;
                    break;
                case "float":
                    _size = 4;
                    break;
                default:
                    System.err.println(_type);
                    break;
            }
        }
        else
        {
            String[] val = _type.split(":");
            _type = val[0]+":";
            try {
                _size = Integer.parseInt(val[1]);
            }
            catch(Exception e)
            {
                _size = 0;
            }
            Main.All_Types.add(_type);
         //   System.out.println(String.format("Structures.Type is %s and Size is %s",_type,_size));
        }
    }
}

File: PW/pwAPI/pwAPI/StructuresElement/Type.cs
using System;

namespace pwApi.StructuresElement
{
    public class Type
    {
        public String get_name()
        {
            return Name;
        }

        public String Name;

        public String get_type()
        {
            return _type;
        }

        public String _type;


        public int Size;

        public Type(String name, String type)
        {
            Name = name;
            _type = type;
            if (!_type.Contains(":"))
            {
                switch (_type)
                {
                    case "int32":
                        break;
                    case "float":
                        break;
                    default:
                        Console.WriteLine(_type);
                        break;
                }
            }
            else
            {
                var val = _type.Split(':');
                _type = val[0] + ":";
                try
                {
                    Size = int.Parse(val[1]);
                }
                catch (Exception e)
                {
                    Size = 0;
                }
            }
        }
    }
}
